include/openmc/plot.h

```diff
@@ -1,15 +1,171 @@
 #ifndef OPENMC_PLOT_H
 #define OPENMC_PLOT_H
 
+#include <map>
+#include <sstream>
+
 #include "hdf5.h"
+#include "openmc/position.h"
+#include "openmc/constants.h"
+#include "openmc/particle.h"
+#include "openmc/xml_interface.h"
 
 namespace openmc {
 
-extern "C" void voxel_init(hid_t file_id, const hsize_t* dims, hid_t* dspace,
+//===============================================================================
+// Global variables
+//===============================================================================
+
+constexpr int RED   = 0;
+constexpr int GREEN = 1;
+constexpr int BLUE  = 2;
+
+extern int PLOT_LEVEL_LOWEST; //!< lower bound on plot universe level
+
+extern std::map<int, int> plot_map; //!< map of plot ids to index
+
+extern int n_plots; //!< number of plots in openmc run
+
+class ObjectPlot;
+extern std::vector<ObjectPlot*> plots; //!< Plot instance container
+
+typedef std::vector<std::vector< std::vector<int>>> ImageData;
+
+enum PLOT_TYPE {
+  SLICE = 1,
+  VOXEL = 2
+};
+
+enum PLOT_BASIS {
+  XY = 1,
+  XZ = 2,
+  YZ = 3
+};
+
+enum PLOT_COLOR_BY {
+  CELLS = 1,
+  MATS = 2
+};
+
+//===============================================================================
+// ObjectColor holds color information for plotted objects
+//===============================================================================
+
+struct ObjectColor {
```

Comment:
I think we can also replace this struct with an array to get some more convenient features:
```c++
typedef std::array<char, 3> Color;
```
(Note that we can use `char` rather than `int` because our values are constrained between 0 and 256.)  You can then define `ImageData` with
```c++
typedef std::vector<std::vector<Color>> ImageData;
```
Something that's nice about this is that I think you can just use the `=` operator in place of the `std::move`s (i.e. `data[x][y] = rgb` if `rgb` is of type `Color`).  It's also more efficient because we aren't storing the information that there are thousands of vectors of size 3 floating around.

Reply:
Class names: Good call. No reason to keep the object specification around anymore.

Color/ImageData: Had something similar in a continuation of this work. :) I'll bring it into this PR and use `std::array`